mesa/agent.py

```diff
@@ -132,16 +133,28 @@ def select(
             filter_func (Callable[[Agent], bool], optional): A function that takes an Agent and returns True if the
                 agent should be included in the result. Defaults to None, meaning no filtering is applied.
             n (int, optional): The number of agents to select. If 0, all matching agents are selected. Defaults to 0.
+            p (float, optional): The fraction of agents to select. If 0.2, it selects 20% of Agents in the AgentSet. Defaults to 1.0.
```

Comment:
p suggests random, which it not really is because it is based on the order in the underlying dict. 

My suggestion is to add something here stating that if you want to ensure randomness, first shuffle in place. Could also be covered by a note.

Reply:
Yeah I'm constantly (almost) forgetting to shuffle before select.

With `n` and `p`, it almost seems like shuffle should be the default. Except of course when you sort first, then you want a sorted number/percentage.

Reply:
Based on this discussion I would suggest renaming it to frac or fraction, based on how its called in pandas or polars, respectively. I would prefer fraction

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sample.html
https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.sample.html

Reply:
Agreed!

Reply:
Done